forge-game/src/main/java/forge/game/staticability/StaticAbility.java

```diff
@@ -63,6 +66,8 @@ public class StaticAbility extends CardTraitBase implements IIdentifiable, Clone
     private final List<Player> ignoreEffectPlayers = Lists.newArrayList();
     private int mayPlayTurn = 0;
 
+    private SpellAbility payingTrigSA;
```

Comment:
you need to update it on:
* StaticAbility.setHostCard
* StaticAbility.copy (copy it too)

with that it should always have the right host (the same as the StaticAbility) even on clones